src/proxy/hdrs/XPACK.cc

```diff
@@ -225,7 +225,7 @@ XpackDynamicTable::XpackDynamicTable(uint32_t size) : _maximum_size(size), _avai
 XpackDynamicTable::~XpackDynamicTable()
 {
   if (this->_entries) {
-    delete this->_entries;
+    ats_free(this->_entries);
```

Comment:
The malloc, for reference, is a few lines above in the constructor here:
https://github.com/apache/trafficserver/pull/11049/files#diff-c9759c90bdcbd117f1d5b745a67412066eff8162ec490431323162d74dcc908cR220

```cpp
  this->_entries = static_cast<struct XpackDynamicTableEntry *>(ats_malloc(sizeof(struct XpackDynamicTableEntry) * size));
```